mainwindow.cpp

```diff
@@ -19,3 +54,529 @@ MainWindow::~MainWindow()
+                                           waypointLongitude,
+                                           waypointAltitude,
+                                           waypointTurnRadius,
+                                           waypointType};
+
+            convertMessage(waypointInfo, MESSAGE_ID_WAYPOINTS);
+        }
+    }
+}
+
+
+void MainWindow::pigoFileChanged(const QString & path)
+{
+   if (allowReading){
+       if (QFile::exists(path)) {
+            watcher->addPath(path);
+        }
+
+      QFile file(path);
+      file.open(QFile::ReadOnly | QFile::Text);
+      QTextStream in(&file);
+      QString text = in.readAll();
+
+
+      std::string current_locale_text = text.toLocal8Bit().constData();
+      json j = json::parse(current_locale_text);
+
+      QString beginTakeoff = QString::number((int)j["beginTakeoff"]);
+      convertMessage(beginTakeoff, MESSAGE_ID_BEGIN_TAKEOFF);
+
+      QString beginLanding = QString::number((int)j["beginLanding"]);
+      convertMessage(beginLanding, MESSAGE_ID_BEGIN_LANDING);
+
+      QString groundHeading = QString::number((float)j["groundCommands"]["heading"]);
+      QString groundLatestDistance = QString::number((float)j["groundCommands"]["latestDistance"]);
+
+      QList<QString> groundInfo = {groundHeading,
+                                   groundLatestDistance
+                                  };
+
+      convertMessage(groundInfo, MESSAGE_ID_GROUND_CMD);
+
+
+      QString gimbalPitch = QString::number((float)j["gimbalCommands"]["pitch"]);
+      QString gimbalYaw = QString::number((float)j["gimbalCommands"]["yaw"]);
+
+      QList<QString> gimbalInfo = {gimbalPitch,
+                                   gimbalYaw
+                                  };
+
+      convertMessage(gimbalInfo, MESSAGE_ID_GIMBAL_CMD);
+
+
+      QString gpsLat = QString::number((int)j["CVGpsCoordinatesOfLandingSpot"]["latitude"]);
+      QString gpsLon = QString::number((int)j["CVGpsCoordinatesOfLandingSpot"]["longitude"]);
+      QString gpsAlt = QString::number((int)j["CVGpsCoordinatesOfLandingSpot"]["altitude"]);
+      QString gpsDOL = QString::number((float)j["CVGpsCoordinatesOfLandingSpot"]["direction of landing"]);
+
+      QList<QString> gpsInfo = {gpsLat,
+                                gpsLon,
+                                gpsAlt,
+                                gpsDOL
+                               };
+      convertMessage(gpsInfo, MESSAGE_ID_GPS_LANDING_SPOT);
+   }
+}
+
+void MainWindow::on_readingButton_clicked()
+{
+    if (allowReading){
+        ui->readingButton->setText("Start Reading");
+        allowReading = !allowReading;
+    }
+    else if (PIGOFilePath != ""){
+        ui->readingButton->setText("Stop Reading");
+        watcher->addPath(PIGOFilePath);
+        allowReading = !allowReading;
+    }
+
+}
+
+void MainWindow::on_pigoBrowseButton_clicked()
+{
+    PIGOFilePath = QFileDialog::getOpenFileName(this, "Open the CV PIGO File", "C:\\", "JSON File (*.json)");
```

Comment:
Add `import <QDir>` to mainwindow.h and replace `C:\\` with `QDir::homePath()`. 

Do the same for the pogi file.